row_cache.cc

```diff
@@ -1342,8 +1342,11 @@ future<> row_cache::do_update(row_cache::external_updater eu, row_cache::interna
         return get_units(_update_sem, 1);
     }).then([this, &eu, &iu] (auto permit) mutable {
       return eu.prepare().then([this, &eu, &iu, permit = std::move(permit)] () mutable {
-        eu.execute();
         [&] () noexcept {
+            // Any error from execute is considered fatal
+            // to enforce exception safety.
+            eu.execute();
```

Comment:
It would be nicer to catch the error, explain what's going on, then abort.

Reply:
ok, I actually had a version that did exactly that and I changed my mind to lean towards brevity.